ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -835,14 +842,33 @@ Result conv2d(
         }
         conv_config.deallocate_activation = true;
     }
+
+    auto output_parallel_config = parallel_config;
+    if(!conv_config.shard_layout.has_value()) {
```

Comment:
this whole block lines 846-862 feels like it should belong in determine_parallel_config function()
is there any reason why it's not there?

Reply:
I needed to add a new argument called output_parallel_config to the conv2d main function. I can put the logic that determines into into another function to clean up.

Reply:
conv_config also gets updated. I don't want to hide that in a different function.